### PR TITLE
Port 0.39 diffs to main

### PR DESCRIPTION
**Pull request by anthony-murphy**
Our 0.39 definitions seem to have diverged from those in main. This change brings missing things to main.

### CODE REVIEW

File: common/lib/container-definitions/src/loader.ts [hunk 1]
```diff
@@ -334,7 +334,7 @@ export interface ILoaderHeader {
     [LoaderHeader.loadMode]: IContainerLoadMode;
     [LoaderHeader.sequenceNumber]: number;
     [LoaderHeader.reconnect]: boolean;
-    [LoaderHeader.version]: string | undefined;
+    [LoaderHeader.version]: string | undefined | null;
```
Comment (anthony-murphy) [inline common/lib/container-definitions/src/loader.ts:337]: @jatgarg is this right? or should it be removed in main?

File: common/lib/container-definitions/src/runtime.ts [hunk 2]
```diff
@@ -111,7 +111,7 @@ export interface IContainerContext extends IDisposable {
     readonly configuration: IFluidConfiguration;
     readonly clientId: string | undefined;
     readonly clientDetails: IClientDetails;
-    readonly storage: IDocumentStorageService;
+    readonly storage: IDocumentStorageService | undefined;
```
Comment (anthony-murphy) [inline common/lib/container-definitions/src/runtime.ts:114]: @jatgarg main shouldn't have undefined correct
```suggestion
    readonly storage: IDocumentStorageService;
```

File: common/lib/container-definitions/src/runtime.ts [hunk 3]
```diff
@@ -121,7 +121,7 @@ export interface IContainerContext extends IDisposable {
     readonly quorum: IQuorum;
     readonly audience: IAudience | undefined;
     readonly loader: ILoader;
-    readonly logger: ITelemetryBaseLogger;
+    readonly logger: ITelemetryLogger;
```
Comment (anthony-murphy) [inline common/lib/container-definitions/src/runtime.ts:124]: @markfields is this correct?

File: common/lib/driver-definitions/src/storage.ts [hunk 4]
```diff
@@ -291,9 +291,13 @@ export interface IDocumentServiceFactory {
...
 
-    // Creates a new document on the host with the provided options. Returns the document service.
+    /**
+     * Creates a new document with the provided options. Returns the document service.
+     * @param createNewSummary - Summary used to create file. If undefined, an empty file will be created and a summary
+     * should be posted later, before connecting to ordering service.
+     */
     createContainer(
-        createNewSummary: ISummaryTree,
+        createNewSummary: ISummaryTree | undefined,
```
Comment (anthony-murphy) [inline common/lib/driver-definitions/src/storage.ts:300]: @wes-carlson this is how it should look correct?

Comment (wes-carlson) [reply common/lib/driver-definitions/src/storage.ts:300]: yep

**Review by wes-carlson — APPROVED** [review]

File: common/lib/container-definitions/src/loader.ts (hunk 1 above)
Comment (anthony-murphy) [inline common/lib/container-definitions/src/loader.ts:337]:
```suggestion
    [LoaderHeader.version]: string | undefined;
```

File: common/lib/container-definitions/src/runtime.ts (hunk 3 above)
Comment (vladsud) [reply common/lib/container-definitions/src/runtime.ts:124]: Nope, ITelemetryBaseLogger is correct one - all compat boundaries should be using simpler and more dump ITelemetryBaseLogger interface.

File: common/lib/container-definitions/src/loader.ts (hunk 1 above)
Comment (vladsud) [reply common/lib/container-definitions/src/loader.ts]: I believe null was deprecated in main.
Here, similar to logging change, the direction should be reverse - find a way to make these changes in 0.39, or bump definitions up (to 0.40) as they are both breaking changes.

File: common/lib/container-definitions/src/runtime.ts (hunk 3 above)
Comment (anthony-murphy) [inline common/lib/container-definitions/src/runtime.ts:124]:
```suggestion
    readonly logger: ITelemetryBaseLogger;
```

File: common/lib/container-definitions/api-report/container-definitions.api.md [hunk 5]
```diff
@@ -161,7 +161,7 @@ export interface IContainerContext extends IDisposable {
     // (undocumented)
     readonly loader: ILoader;
     // (undocumented)
-    readonly logger: ITelemetryBaseLogger;
+    readonly logger: ITelemetryLogger;
```
Comment (anthony-murphy) [inline common/lib/container-definitions/api-report/container-definitions.api.md:164]:
```suggestion
    readonly logger: ITelemetryBaseLogger;
```

File: common/lib/container-definitions/api-report/container-definitions.api.md [hunk 6]
```diff
@@ -33,7 +33,7 @@ import { ISignalClient } from '@fluidframework/protocol-definitions';
 import { ISignalMessage } from '@fluidframework/protocol-definitions';
 import { ISnapshotTree } from '@fluidframework/protocol-definitions';
 import { ISummaryTree } from '@fluidframework/protocol-definitions';
-import { ITelemetryBaseLogger } from '@fluidframework/common-definitions';
+import { ITelemetryLogger } from '@fluidframework/common-definitions';
```
Comment (anthony-murphy) [inline common/lib/container-definitions/api-report/container-definitions.api.md:36]:
```suggestion
import { ITelemetryBaseLogger} from '@fluidframework/common-definitions';
```

File: common/lib/container-definitions/src/runtime.ts [hunk 7]
```diff
@@ -3,7 +3,7 @@
  * Licensed under the MIT License.
  */
 
-import { ITelemetryBaseLogger, IDisposable } from "@fluidframework/common-definitions";
+import { ITelemetryLogger, IDisposable } from "@fluidframework/common-definitions";
```
Comment (anthony-murphy) [inline common/lib/container-definitions/src/runtime.ts:6]:
```suggestion
import { ITelemetryBaseLogger, IDisposable } from "@fluidframework/common-definitions";
```

File: common/lib/container-definitions/api-report/container-definitions.api.md [hunk 8]
```diff
@@ -33,7 +33,7 @@ import { ISignalClient } from '@fluidframework/protocol-definitions';
 import { ISignalMessage } from '@fluidframework/protocol-definitions';
 import { ISnapshotTree } from '@fluidframework/protocol-definitions';
 import { ISummaryTree } from '@fluidframework/protocol-definitions';
-import { ITelemetryBaseLogger } from '@fluidframework/common-definitions';
+import { ITelemetryBaseLogger} from '@fluidframework/common-definitions';
```
Comment (anthony-murphy) [inline common/lib/container-definitions/api-report/container-definitions.api.md:36]:
```suggestion
import { ITelemetryBaseLogger } from '@fluidframework/common-definitions';
```

File: common/lib/container-definitions/src/loader.ts (hunk 1 above)
Comment (anthony-murphy) [reply common/lib/container-definitions/src/loader.ts]: yup. that sounds right. i'm looking into what bumping to 0.40 looks like, so just reviewing changes. pr seemed the best way to get feedback/answers to understand the breaks.

**Pull request merged**